### PR TITLE
Bugfix/1369 date parsing fixes

### PR DESCRIPTION
**Pull request by davidnich**

### CODE REVIEW

File: lib/qore_date_private.cpp [hunk 1]
```diff
@@ -279,6 +331,12 @@ void qore_absolute_time::set(const char* str, const AbstractQoreZoneInfo* n_zone
       return;
    }
 
+   if (xsink && second > 59) {
```
Comment (omusil24) [inline lib/qore_date_private.cpp:334]: Shouldn't this be 60 to account for possible leap seconds?

Comment (gamato) [reply lib/qore_date_private.cpp:334]: That's something I noticed too, but then there's another question whether it should be done like with days where Qore actually checks how many days are allowed in particular month.  But I'm not sure what the rules with leap seconds are, likely they can be inserted at any time (though usually at the very end of the year).

Comment (omusil24) [reply lib/qore_date_private.cpp:334]: Unfortunately leap seconds are inserted quite randomly depending on Earth's rotation, therefore this check would not be very easy.

Comment (davidnich) [reply lib/qore_date_private.cpp:334]: Currently Qore doesn't support leap seconds.  according to what I found, there are only conventions regarding when a leap second is added (normally the end of the year or the end of June) but no rules.  Additionally, there is no long-term schedule for leap seconds, rather they are decided relatively shortly in advance on an ad-hoc basis based on the actual difference in the earth's rotation and atomic time.

So we can support leap seconds by changing the `59` to a `60` here, but in my opinion, it's not possible to verify them (at least not for future dates).

Comment (gamato) [reply lib/qore_date_private.cpp:334]: I agree.  Maybe as an extra precaution we can check if the leap second is the last second of the day, but maybe it does not really matter.  So I think it's OK to merge this.

**Comment by omusil24** [comment]
Ok, so since there are otherwise no issues, I'm merging this. The leap second fix can be done anytime in the future quite easily.

**Pull request merged**